org.eclipse.lsp4e/src/org/eclipse/lsp4e/callhierarchy/CallHierarchyContentProvider.java

```diff
@@ -86,42 +90,38 @@ public void inputChanged(final Viewer viewer, final Object oldInput, final Objec
 			rootItems = null;
 
 			IDocument document = viewInput.getDocument();
-			if (document != null) {
-				try {
-					initialise(document, viewInput.getOffset());
-				} catch (BadLocationException e) {
-					handleRootError();
-				}
-			} else {
+			try {
+				initialise(document, viewInput.getOffset());
+			} catch (Exception e) {
 				handleRootError();
 			}
 		} else {
 			handleRootError();
 		}
-
 	}
 
-	private void initialise(final @NonNull IDocument document, final int offset) throws BadLocationException {
+	private void initialise(final IDocument document, final int offset) throws BadLocationException {
 		LanguageServerDocumentExecutor executor = LanguageServers.forDocument(document)
 				.withCapability(ServerCapabilities::getCallHierarchyProvider);
 		if (!executor.anyMatching()) {
 			handleRootError();
 			return;
 		}
-		CallHierarchyPrepareParams prepareParams = LSPEclipseUtils.toCallHierarchyPrepareParams(offset, document);
```

Comment:
To keep it consistent with the rest of the elements, I would leave it in `LSPEclipseUtils`